lib/system/gc_ms.nim

```diff
@@ -305,20 +305,22 @@ proc newObjRC1(typ: PNimType, size: int): pointer {.compilerRtl.} =
   when defined(memProfiler): nimProfile(size)
 
 when not defined(nimSeqsV2):
+  {.push overflowChecks: on.}
```

Comment:
pre-existing behavior but why do we need to force overflowChecks, overriding -d:danger here?
seems like it could impact performance

Reply:
Because the last thing you want to have is an undetected wrap-around when computing allocation sizes. It's crucial for security.